src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImpl.java

```diff
@@ -1,70 +1,91 @@
 package uk.gov.companieshouse.api.testdata.service.impl;
 
-import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import uk.gov.companieshouse.api.testdata.exception.DataException;
 import uk.gov.companieshouse.api.testdata.model.entity.CompanyPscs;
+import uk.gov.companieshouse.api.testdata.model.entity.Identification;
 import uk.gov.companieshouse.api.testdata.model.entity.Links;
+import uk.gov.companieshouse.api.testdata.model.entity.NameElements;
 import uk.gov.companieshouse.api.testdata.model.rest.CompanySpec;
 import uk.gov.companieshouse.api.testdata.repository.CompanyPscsRepository;
 import uk.gov.companieshouse.api.testdata.service.DataService;
 import uk.gov.companieshouse.api.testdata.service.RandomService;
 
+import java.time.Clock;
 import java.time.Instant;
-import java.time.LocalDate;
-import java.time.ZoneId;
 import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Optional;
+import java.util.*;
 
 @Service
 public class CompanyPscsServiceImpl implements DataService<CompanyPscs> {
 
+    protected static final String[] NATURES_OF_CONTROL = {"ownership-of-shares-25-to-50-percent", "ownership-of-shares-50-to-75-percent",
+        "ownership-of-shares-75-to-100-percent", "ownership-of-shares-25-to-50-percent-as-trust", "ownership-of-shares-50-to-75-percent-as-trust"};
     private static final int ID_LENGTH = 10;
     private static final int SALT_LENGTH = 8;
+    public static final String WALES = "Wales";
 
-    @Autowired
     private RandomService randomService;
 
-    @Autowired
     private CompanyPscsRepository repository;
 
+    private Clock clock;
+
+    @Autowired
+    public CompanyPscsServiceImpl(RandomService randomService, CompanyPscsRepository repository, Clock clock) {
+        this.randomService = randomService;
+        this.repository = repository;
+        this.clock = clock;
+    }
 
     @Override
     public CompanyPscs create(CompanySpec spec) throws DataException {
```

Comment:
Possible refactoring of this long method. Seems logical to introduce methods like `setLocation` for all the address-related lines, `setNaturesOfControl` for NOC lines, etc.